exception.ts

```diff
@@ -226,7 +226,7 @@ export type ErrorPredicate<E> =
   | ErrorConstructorPredicate<E>
   | ErrorPredicateFunction<E>
   | ErrorUnionPredicate<E>;
-export type ErrorConstructorPredicate<out E> = new (...args: unknown[]) => E;
+export type ErrorConstructorPredicate<out E> = new () => E;
```

Comment:
This in turn invalidates error constructors with required arguments. Could you add a test for that case and fix the type accordingly (namely `...args: any[]` with lint suppression)?